components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/security/APIAuthenticationHandler.java

```diff
@@ -753,9 +753,35 @@ private void handleAuthFailure(MessageContext messageContext, APISecurityExcepti
             Map<String, String> headers =
                     (Map) axis2MC.getProperty(org.apache.axis2.context.MessageContext.TRANSPORT_HEADERS);
             if (headers != null) {
-                headers.put(HttpHeaders.WWW_AUTHENTICATE, getAuthenticatorsChallengeString() +
-                        " error=\"invalid_token\"" +
-                        ", error_description=\"The provided token is invalid\"");
+                if (APIConstants.API_TYPE_MCP.equalsIgnoreCase(apiType)) {
+                    // Derive the outward facing host and port from host header
+                    String hostHeader = headers.get(APIMgtGatewayConstants.HOST);
+                    Pattern validHostHeaderPattern = Pattern.compile("^[A-Za-z0-9][A-Za-z0-9.-]*(:\\d{1,5})?$");
+
+                    if (StringUtils.isBlank(hostHeader) || !validHostHeaderPattern.matcher(hostHeader).matches()) {
+                        log.debug("Missing or malformed host header in request.Extracting host header form config.");
```

Comment:
`request.Extracting` needs to be fixed to `request. Extracting`

Reply:
This logic is changed